aip/general/0162.md

```diff
@@ -27,199 +27,174 @@ to the version of an API as a whole.
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
       type: "library.googleapis.com/Book"
     }];
+  string tag = 2 [(google.api.field_behavior) = OPTIONAL];
 }
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+- The method **must** use the `POST` HTTP verb.
+- The method **should** return a long running operation or resource revision.
+- The request message **must** include the `name` field.
+- The resource ID of the resource revision **should** be a string and contain a
+  short, automatically-generated random string. A good rule of thumb is the last
+  eight characters of a [UUID version 4 (random)](UUID4).
+- `tag` field is optional. If provided, revision created will have the tag set.
+
+### Getting a revision
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+Services `must` implement the standard `Get` method ([AIP-131][])
 
 ### Tagging revisions
 
+There are two types of tags: server and user specified.
+
+#### Server specified
+
+Services **may** reserve specific tags to be server-specified (e.g. `current`,
+`latest`), which are read-only and managed by the service.
+
+- If a `latest` tag exists, it **must** represents the most recent created
+  revision.
+ The content of `publishers/{publisher}/books/{book}/revisions/latest` and
+  `publishers/{publisher}/books/{book}` can differ. Because the latest revision
+  may be different from the current state of the resource.
+
+```
+GET /v1/publishers/{publisher}/books/{book}/revisions/{reserved_keyword}
+```
+
+#### User specified
+
+User specified tags are tags provided by the user when tagging specific resource
+revision.
+
 APIs implementing resource revisions **may** provide a mechanism for users to
 tag a specific revision with a user provided name by implementing a "Tag
 Revision" custom method:
 
 ```proto
 rpc TagBookRevision(TagBookRevisionRequest) returns (Book) {
   option (google.api.http) = {
-    post: "/v1/{name=publishers/*/books/*}:tagRevision"
+    post: "/v1/{name=publishers/*/books/*/revisions/*}:tag"
     body: "*"
   };
 }
 ```
 
 ```proto
 message TagBookRevisionRequest {
-  // The name of the book to be tagged, including the revision ID.
+  // The name of the book revision to be tagged.
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
+      type: "library.googleapis.com/BookRevision"
     }];
 
-  // The tag to apply.
-  // The tag should be at most 40 characters, and match `[a-z][a-z0-9-]{3,38}[a-z0-9]`.
+  // The tag to apply. The tag should be at most 40 characters, and match
```

Comment:
I think we spoke about this already, but tags are almost resource IDs as this point. This whole request could be reformatted in terms of a create.